File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_AssetVersion.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_AssetVersion
*/
public class t_AssetVersion : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_verionAsset;
	public uint	m_versionClient;
	public ulong	m_uMd5a;
	public ulong	m_uMd5b;
	public t_AssetVersionOne[]	m_aVersion;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_AssetVersion() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_verionAsset = 0;
		m_versionClient = 0;
		m_uMd5a = 0UL;
		m_uMd5b = 0UL;
		m_aVersion = new t_AssetVersionOne[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_verionAsset = cVariable.getU32();
			m_versionClient = cVariable.getU32();
			m_uMd5a = cVariable.getU64();
			m_uMd5b = cVariable.getU64();
			{
				int n = (int) cVariable.getU16();
				if (n > 65535) {
					cVariable.error("array size error in m_aVersion");
					return false;
				}
				m_aVersion = new t_AssetVersionOne [n];
				for(int i = 0;i < n;++i) {
					m_aVersion[i] = new t_AssetVersionOne();
					if (!m_aVersion[i].read(cVariable)) return false;
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_verionAsset);
		cVariable.put(m_versionClient);
		cVariable.put(m_uMd5a);
		cVariable.put(m_uMd5b);
		{	
				if (m_aVersion.Length > 65535) {
					cVariable.error("array size error in m_aVersion");
					return false;
				}
			int n = m_aVersion.Length;
			cVariable.put((ushort) n);
			for(int i = 0;i < n;++i) {
				if (!m_aVersion[i].write(cVariable)) return false;

			}
		}
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_AssetVersion.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_AssetVersionOne import	t_AssetVersionOne
#/==========================================================================
#*!
#	@brief	t_AssetVersion
#/
class t_AssetVersion:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_verionAsset = 0	#U32
	m_versionClient = 0	#U32
	m_uMd5a = 0	#U64
	m_uMd5b = 0	#U64
	m_aVersion = []	#t_AssetVersionOne
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_verionAsset = 0
		self.m_versionClient = 0
		self.m_uMd5a = 0
		self.m_uMd5b = 0
		self.m_aVersion = []

	def read(self,cVariable):
		self.m_verionAsset = cVariable.getU32()
		self.m_versionClient = cVariable.getU32()
		self.m_uMd5a = cVariable.getU64()
		self.m_uMd5b = cVariable.getU64()
		n = cVariable.getU16()
		if (n > 65535) :
			cVariable.error('array size error in m_aVersion')
			return False
		self.m_aVersion = [None]*n
		for i in range(n):
			self.m_aVersion[i] = t_AssetVersionOne()
			if (not self.m_aVersion[i].read(cVariable)):
				return False
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_verionAsset)
		cVariable.putU32(self.m_versionClient)
		cVariable.putU64(self.m_uMd5a)
		cVariable.putU64(self.m_uMd5b)
		if (len(self.m_aVersion) > 65535) :
			cVariable.error('array size error in m_aVersion')
			return False
		n = len(self.m_aVersion)
		cVariable.putU16(n)
		for i in range(n):
			if (not self.m_aVersion[i].write(cVariable)):
				return False
		return True

